Improve contributing documentation (#49)
label-sleuth/label-sleuth.org: docs/contributing.md

git checkout -b {your-branch-name}
```

### 5. Update your branch

When working on a branch, it is possible that extra functionality has been added to the
main branch of the repository, which means that your fork is outdated. In order to update your branch run:

```sh
git pull --rebase upstream main
```

This command will apply all the commits that are on upstream/main but not on your local branch.

> Addind the `--rebase` flag lets the commit history to remain clean (i.e. no merge commits). The downside

Reviewer: ```suggestion
Adding the `--rebase` flag lets the commit history remain clean (i.e., without merge commits). The downside
```

> of rebasing is that is will change the hash id of your own commits, which means that, in order to update

Reviewer: ```suggestion
of rebasing is that it will change the hash id of your own commits, which means that, in order to update
```

> your fork on Github, you'll have to do a forced push:

Reviewer: ```suggestion
your fork on Github, you will have to do a forced push:
```

```sh
git push --force-with-lease origin {your-branch-name}
```

> _note: --force-with-lease is used here instead of --force to ensure that, if new commits have been added

Reviewer: ```suggestion
_Note: `--force-with-lease` is used here instead of `--force` to ensure that, if new commits have been added
```

> to your remote branch (i.e. you are working on the same branch with someone else) the push command will fail._

Reviewer: ```suggestion
to your remote branch (e.g., if you are working on the same branch with someone else) the push command will fail._
```
